src/main/docker/gitlab-jenkins-mysql.yml

```diff
@@ -9,27 +9,33 @@ services:
             - ./gitlab/gitlab-local-setup.sh:/gitlab-local-setup.sh
         environment:
             - GITLAB_OMNIBUS_CONFIG=prometheus_monitoring['enable'] = false; gitlab_rails['gitlab_shell_ssh_port'] = 2222; gitlab_rails['monitoring_whitelist'] = ['0.0.0.0/0']
+            - GITLAB_ROOT_PASSWORD
         ports:
             - "2222:22"
             - "8081:80"
         hostname: gitlab
         networks:
-            artemis:
+            artemis-gitlab:
                 ipv4_address: 172.33.0.2
     jenkins:
         build: jenkins
+        user: root
         volumes:
             - artemis-jenkins-data:/var/jenkins_home
-            - /var/run/docker.sock:/var/run/docker.sock
-            - /usr/local/bin/docker:/usr/bin/docker
+            - ./jenkins/jenkins-disable-csrf.groovy:/var/jenkins_home/init.groovy # Disable CSRF token
+            - ./jenkins/master.key:/var/jenkins_home/master.key # Preset master key to use pre-generated secrets
             - ./jenkins/jenkins-casc-config.yml:/usr/share/jenkins/ref/jenkins-casc-config.yml
+            - /var/run/docker.sock:/var/run/docker.sock
+            - /usr/bin/docker:/usr/bin/docker:ro
+            - /usr/bin/com.docker.cli:/usr/bin/com.docker.cli
         ports:
             - "8082:8080"
         hostname: jenkins
         environment:
             - CASC_JENKINS_CONFIG=/usr/share/jenkins/ref/jenkins-casc-config.yml
+            - JAVA_OPTS
         networks:
-            artemis:
+            artemis-gitlab:
                 ipv4_address: 172.33.0.3
```

Comment:
Why do you fix the IP addresses here?

Reply:
Some people had issues with the "normal" networks iirc and it makes debugging other dev's setup a bit easier as the ip addresses are known (e.g. if the networking between the containers is not working as expected).

But this is not really a hard requirement.